framework/assets/yii.gridView.js

```diff
@@ -85,25 +85,40 @@
             var $grid = $(this), event;
             var settings = gridData[$grid.attr('id')].settings;
             var data = {};
-            $.each($(settings.filterSelector).serializeArray(), function () {
-                data[this.name] = this.value;
-            });
+            var dataForm = [];
 
+            // get the query params from the request url
             $.each(yii.getQueryParams(settings.filterUrl), function (name, value) {
-                if (data[name] === undefined) {
-                    data[name] = value;
+                data[name.replace(/\[[0-9]+\]/i, '[]')] = value;
```

Comment:
Yes I know that. The problem is that we cannot be sure that the values from `name` will have numbers or not. For example the paginaton adds them automatically. I think that we don't have to modify that part because we don't have any problem with the pagination. So we remove all numbers to make the comparison. 
I don't like the solution either but I haven't think of a better one